File: Coder Army/SOLID/DIP(dependencyInversionPrinciple)/DIP_violated.cpp
#include <iostream>
using namespace std;

class MySql
{
public:
    void saveToSql(string data)
    {
        cout << "Insert into users value ('" << data << "')" << endl;
    }
};

class Mongodb
{
public:
    void saveToMongodb(string data)
    {
        cout << "Insert into db.company.users value ('" << data << "')" << endl;
    }
};

class UserService
{
private:
    MySql sqlDB;
    Mongodb mongoDB;

public:
    void storeUserToSQL(string user)
    {
        sqlDB.saveToSql(user);
    }
    void storeUserToMongoDB(string user)
    {
        mongoDB.saveToMongodb(user);
    }
};

int main()
{
    UserService service;
    service.storeUserToMongoDB("Kuldeep");
    service.storeUserToSQL("Kumar");

    return 0;
}

File: Coder Army/SOLID/DIP(dependencyInversionPrinciple)/Javascript code/DIP_violated.js
class MySql {
  saveToSql(data) {
    console.log(`Insert into users value ('${data}')`);
  }
}

class MongoDB {
  saveToMongodb(data) {
    console.log(`Insert into db.company.users value ('${data}')`);
  }
}

class UserService{
  constructor(){
    this.sqlDB= new MySql() // MySql sqlDB
    this.mongoDB= new MongoDB() // Mongodb= mmongoDB
  }

  storeUserToSQL(user){
    this.sqlDB.saveToSql(user)
  }
  storeUserToMongoDB(user){
    this.mongoDB.saveToMongodb(user)
  }
}

const service= new UserService()
service.storeUserToMongoDB("Ram")
service.storeUserToSQL("shyam")